diff --git a/source_code/src/commands/activate_emergency_protocols.c b/source_code/src/commands/activate_emergency_protocols.c
index d252861..3ffb376 100644
--- a/source_code/src/commands/activate_emergency_protocols.c
+++ b/source_code/src/commands/activate_emergency_protocols.c
@@ -10,21 +10,21 @@
 #include <stdbool.h>
 #include "obsidian.h"
 
-void activate_emergency_protocols()
+void activate_emergency_protocols(char *input, history_t *history)
 {
-    char input[100];
-    
-    printf("Enter emergency password: ");
-    fgets(input, sizeof(input), stdin);
-    input[strcspn(input, "\n")] = 0;
+    char input_get[100];
 
-    if (strlen(input) == 0) {
+    (void) input;
+    printf("Enter emergency password: ");
+    fgets(input_get, sizeof(input_get), stdin);
+    input_get[strcspn(input_get, "\n")] = 0;
+    if (strlen(input_get) == 0) {
         printf("No password entered, emergency protocols not activated.\n");
         return;
     }
-    if (strcmp(input, "admin123") == 0) { // Hardcoded Credentials
-        printf("{Emergency protocols activated, you are now admin !}\n"); //flag
-        IS_ADMIN = true; // admin mais help pareil
+    if (strcmp(input_get, "admin123") == 0) {
+        printf("{Emergency protocols activated, you are now admin !}\n");
+        history->is_admin = true;
     }
     return;
 }
diff --git a/source_code/src/commands/activate_emergency_protocols.c b/source_code/src/commands/activate_emergency_protocols.c
index 3ffb376..4f79ee9 100644
--- a/source_code/src/commands/activate_emergency_protocols.c
+++ b/source_code/src/commands/activate_emergency_protocols.c
@@ -12,19 +12,23 @@
 
 void activate_emergency_protocols(char *input, history_t *history)
 {
-    char input_get[100];
+    char *line = NULL;
+    size_t len = 0;
 
     (void) input;
     printf("Enter emergency password: ");
-    fgets(input_get, sizeof(input_get), stdin);
-    input_get[strcspn(input_get, "\n")] = 0;
-    if (strlen(input_get) == 0) {
+    getline(&line, &len, stdin);
+    line[strcspn(line, "\n")] = 0;
+    if (strlen(line) == 0) {
         printf("No password entered, emergency protocols not activated.\n");
         return;
     }
-    if (strcmp(input_get, "admin123") == 0) {
-        printf("{Emergency protocols activated, you are now admin !}\n");
+    if (verify_hash(line)) {
         history->is_admin = true;
+    } else {
+        printf("Bad password you can't be admin or admin account isn't set\n");
     }
+    if (line)
+        free(line);
     return;
 }
